tests/unit/deploy/test_swagger.py

```diff
@@ -41,6 +41,96 @@ def test_can_produce_doc_for_method(sample_app, swagger_gen):
     # Same for x-amazon-apigateway-integration.
 
 
+def test_can_produce_doc_for_no_docstring(sample_app, swagger_gen):
+    @sample_app.route('/method')
+    def method():
+        pass
+
+    doc = swagger_gen.generate_swagger(sample_app)
+    view_config = doc['paths']['/method']['get']
+    assert 'summary' not in view_config
+    assert 'description' not in view_config
+
+
+def test_can_produce_doc_for_single_docstring(sample_app, swagger_gen):
+    @sample_app.route('/method1')
+    def method1():
+        """Single line method summary"""
+        pass
+
+    @sample_app.route('/method2')
+    def method2():
+        '''Single line method summary'''
+        pass
+
+    @sample_app.route('/method3')
+    def method3():
+        """
+        Single line method summary
+        """
+        pass
+
+    doc = swagger_gen.generate_swagger(sample_app)
+    for method in [1, 2, 3]:
```

Comment:
How about parameterizing these? https://docs.pytest.org/en/latest/example/parametrize.html

Reply:
yes, working on it...

Reply:
So I've used parameterized tests before and love them for static data sets, and tried refactoring the code for this case. I think it reads much cleaner as it is though. If I move to parameterized, then each of the methods get defined outside the test, then I have the block of function data mapped to a ```method``` variable. In the end I thought it was less readable, though you do get cleaner reporting if one of the sub-tests fail.

Reply:
It was mainly the cleaner reporting that I was going for. I was thinking these would be clearer as separate tests but didn't want the repeated logic since they were all so separate.

Reply:
I could still break into separate tests if you prefer, but I think the parameterization didn't look right to me.

Reply:
Is this what is holding up the PR approval? I will complete this work if you want, just want to be sure that's all.